### PR TITLE
Google API Integration 

### LINKED ISSUE #33: Google Geocoding Service - Due EOD 4/12
As a Developer, 
When I submit a request to either the `/recommendation` or `/random` endpoints, 
Using a physical address instead of Long-Lat coordinates, 
The endpoints will submit a request to the Google Geocoding API to retrieve the Long-Lat coordinates, 
Which can then be used in the subsequent Yelp API request to return a Restaurant recommendation.

### LINKED ISSUE #36: Figure out why same restaurant is returned with each API call
Unless there is only 1 restaurant for that category, we should be seeing different restaurants as a result of hitting the /api/vi/recommendations endpoint, but we are getting the same restaurant back every time. We need to see what's going on here.

### PR DESCRIPTION
**Pull request by linda-le1**
## What functionality does this accomplish?
closes #33 
closes #36 

**Description:**
Fixes bug where same restaurant returned each time. Integrates Google API to return long/lat if user does not share current location. 

## What did you struggle on to complete?
Nothing out of the ordinary - just used the debugger to ensure variables and what I was passing over was what I needed. Definitely need more sad path testing and logic to handle that. 

### Test Coverage Percentage: x%
- [ ] No Tests have been changed
- [X] New Tests have been implemented
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened):

## Checklist:
- [ ] My code has no unused/commented out code
- [X] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code
- [ ] I have partially tested my code (please explain why):

## Helpful Resources:
* Resource link AND small description of info gathered/learned

## Review Requests(optional):
I've identified a few places we should add in logic to account for sad paths. Let me know if what your thoughts are! 

## Please include a gif of how you feel about this branch:
giphy

### CODE REVIEW

File: base_app/base.py [hunk 1]
```diff
@@ -21,11 +22,23 @@ def get_random():
...
 
-
 ## Helper Methods ##
 
+def has_address(request_args):
+    params = { 'open_now': True, 'term': 'food'}
+    google= GoogleService()
+    location = {}
+    new_address = request_args['address'].replace(",", "+")
+    address = location.update({'address': f'{new_address}'})
```
Comment (freeheeling) [inline base_app/base.py]: Is this address variable being used anywhere or does the location just need to be modified with the formatted address?

File: base_app/yelp_service.py [hunk 2]
```diff
@@ -23,13 +24,14 @@ def get_restaurant(self, params):
         url = 'https://api.yelp.com/v3/businesses/search'
         response = self.connection(url, params)
         all = response.json()
+        ids = {}
         for x in all['businesses']:
             if x.get('price') != None:
-                ids = []
-                ids.append(x['id'])
+                ids.update({x['id']: x['distance']})
```
Comment (freeheeling) [inline base_app/yelp_service.py]: Can you briefly explain what this line is doing? Is `distance` used to ensure a restaurant is within a certain proximity?

File: tests/recommendations_endpoint_test.py [hunk 3]
```diff
@@ -136,6 +136,58 @@ def test_user_selects_price_preference_4_returns_successful_response():
...
+    assert 'cuisine' in response_body['data']['attributes']
+    assert 'name' in response_body['data']['attributes']
+    assert 'location' in response_body['data']['attributes']
+    assert 'phone' in response_body['data']['attributes']
+    assert 'display_phone' in response_body['data']['attributes']
+    assert 'rating' in response_body['data']['attributes']
+    assert 'price' in response_body['data']['attributes']
+    assert response_body['data']['attributes']['is_closed'] == False
+
+
```
Comment (freeheeling) [inline tests/recommendations_endpoint_test.py:190]: Should there also be a test for full address, including zip code?

File: base_app/base.py [hunk 4]
```diff
@@ -21,11 +22,23 @@ def get_random():
     params = format_params(request.args)
     return service.get_recommendation(params )
 
-
 ## Helper Methods ##
 
+def has_address(request_args):
+    params = { 'open_now': True, 'term': 'food'}
+    google= GoogleService()
```
Comment (freeheeling) [inline base_app/base.py]: Add a space before `=`.

**Review by freeheeling — APPROVED** [review]
Commented with a couple of questions. Also, ecrypted Google API key should be added to `travis.yml`, otherwise the service won't be available in staging/production.

File: tests/recommendations_endpoint_test.py (hunk 3 above)
Comment (linda-le1) [reply tests/recommendations_endpoint_test.py:190]: We can definitely add a test in for this!

File: base_app/yelp_service.py (hunk 2 above)
Comment (linda-le1) [reply base_app/yelp_service.py]: Thanks for catching that! I've taken it out.

File: base_app/base.py (hunk 4 above)
Comment (linda-le1) [reply base_app/base.py]: Fixed!

**Review by freeheeling — APPROVED** [review]
Thank you for addressing my questions and making updates where necessary.

**Pull request merged**